tests/algorithms/test_optimizers.py

```diff
@@ -66,14 +69,18 @@ def test_callback_optimizers(optimizer):
     parameter2 = torch.tensor([3.14], requires_grad=True)
     parameters = [parameter1, parameter2]
 
-    # maximum number of iterations
-    max_iter = 10
-
     # callback function; if the function is called during the iterations, the
     # test is successful
     def callback(optimizer_status: OptimizerStatus) -> None:
         _, _ = optimizer_status['iteration_number'], optimizer_status['solution'][0]
         assert True
 
     # run optimizer
-    _ = optimizer(rosen_brock, parameters, max_iter=max_iter, callback=callback)
+    if optimizer == lbfgs:
```

Comment:
either remove the max_iterations/n_iterations argument here (it does not matter) (preferred)
OR just supply "10" without naming the argument
OR do it optimizer_kwargs in the parameterize, as done in the test above (least favourite)

Reply:
-> dont have this if here.